craft_grammar/models.py

```diff
@@ -50,10 +53,77 @@ def _grammar_append(cls, entry: List, item: Any) -> None:
             _mark_and_append(entry, {key: cls.validate(value)})
 
 
-_GrammarType = Dict[str, Any]
+# Public types for grammar-enabled attributes
+class GrammarBool(_GrammarBase):
+    """Grammar-enabled bool field."""
 
+    __root__: Union[bool, _GrammarType]
 
-# Public types for grammar-enabled attributes
+    @classmethod
+    @overrides
+    def validate(cls, entry):
+        # GrammarInt entry can be a list if it contains clauses
+        if isinstance(entry, list):
+            new_entry = []
+            for item in entry:
+                if _is_grammar_clause(item):
+                    cls._grammar_append(new_entry, item)
+                else:
+                    raise TypeError(f"value must be a list of bool: {entry!r}")
+            return new_entry
+
+        if isinstance(entry, bool):
+            return entry
+
+        raise TypeError(f"value must be a bool: {entry!r}")
+
+
+class GrammarInt(_GrammarBase):
+    """Grammar-enabled integer field."""
+
+    __root__: Union[int, _GrammarType]
+
+    @classmethod
+    @overrides
+    def validate(cls, entry):
+        # GrammarInt entry can be a list if it contains clauses
+        if isinstance(entry, list):
+            new_entry = []
+            for item in entry:
+                if _is_grammar_clause(item):
+                    cls._grammar_append(new_entry, item)
+                else:
+                    raise TypeError(f"value must be a list of integer: {entry!r}")
+            return new_entry
+
+        if isinstance(entry, int):
+            return entry
+
+        raise TypeError(f"value must be a integer: {entry!r}")
+
+
+class GrammarFloat(_GrammarBase):
+    """Grammar-enabled float field."""
```

Comment:
Remind me, what fields are float?

Reply:
After a quick search, seems no current plugins use float. Do you want me remove it?

Reply:
I don't have a strong preference.  I lean towards removing it since it won't be used today.

On the other hand, if you keep it, then it will be ready for use in the future.

Reply:
Seems harmless to me, if we want to add it back later we need to do 2 releases. It seems better to keep it.